packages/vow/src/rejection-tracker.js

```diff
@@ -25,73 +25,73 @@ const makeUnhandledRejection = (rejected, upgradedPK) => {
+  /** @type {WeakMap<PassableCap, () => void>} */
   const vowToCancelUnhandledRejection = new WeakMap();
 
-  const makeVowRejectionTrackerKit = zone.exoClassKit(
-    'VowRejectionTrackerKit',
-    VowRejectionTrackerKitI,
-    () => ({}),
+  const upgradeRejectionWatcher = zone.exo(
+    'UpgradeRejectionWatcher',
+    M.interface('UpgradeRejectionWatcher', {
+      onRejected: M.call(M.raw(), M.raw()).returns(),
+    }),
     {
-      public: {
-        handle(vowV0) {
-          const cancel = vowToCancelUnhandledRejection.get(vowV0);
-          if (!cancel) {
-            throw Fail`vow ${vowV0} already handled`;
-          }
-          cancel();
-        },
-        /**
-         *
-         * @param {VowPayloadV0} vowV0
-         * @param {any} reason
-         * @param {Promise<void>} [rejected]
-         */
-        reject(vowV0, reason, rejected = Promise.reject(reason)) {
-          const { upgradeRejectionWatcher } = this.facets;
-          !vowToCancelUnhandledRejection.has(vowV0) ||
-            Fail`vow ${vowV0} already rejected`;
-
-          // Register a never-resolved native promise with liveslots, so it
-          // can reject on upgrade.
-          const upgradedPK = makePromiseKit();
-          zone.watchPromise(
-            upgradedPK.promise,
-            upgradeRejectionWatcher,
-            reason,
-          );
-
-          // Save the cancel function.
-          const cancel = makeUnhandledRejection(rejected, upgradedPK);
-          vowToCancelUnhandledRejection.set(vowV0, cancel);
-        },
-      },
-      upgradeRejectionWatcher: {
-        onRejected(upgradeReason, baseReason) {
-          if (baseReason instanceof Error) {
-            assert.note(baseReason, X`upgraded: ${upgradeReason}`);
-          }
-          throw baseReason;
-        },
+      onRejected(upgradeReason, baseReason) {
+        const reason = assert.error(
+          X`VOW_REJECTION ${baseReason} not handled before upgrade ${upgradeReason}`,
+        );
+        // console.warn(reason);
+        throw reason;
       },
     },
   );
 
-  const makeVowRejectionTracker = () => makeVowRejectionTrackerKit().public;
-  return makeVowRejectionTracker;
+  const vowRejectionTracker = harden({
+    /**
+     * @param {PassableCap} vowCap
+     */
+    handle(vowCap) {
+      const cancel = vowToCancelUnhandledRejection.get(vowCap);
+      if (!cancel) {
+        console.warn(
+          X`Now handling a VOW_REJECTION from a prior incarnation of ${vowCap}`,
+        );
+        return;
+      }
+      vowToCancelUnhandledRejection.set(vowCap, sink);
+      cancel();
+    },
+    /**
+     * @param {PassableCap} vowCap
+     * @param {any} reason
+     */
+    reject(vowCap, reason) {
+      if (vowToCancelUnhandledRejection.has(vowCap)) {
+        return;
+      }
+
+      harden(reason);
+      const baseReason = zone.isStorable(reason)
+        ? reason
+        : assert.error(`Vow rejection reason was not stored: ${reason}`);
```

Comment:
I think we should be more resilient to `reason` not coercing to a string. Is that something `X` should help with?

Reply:
Tried now, and definitely works.  I didn't use it before because of some of my confusion around what kind of errors are storable.  Now I remember that `X`'s exposing values from the template literal to the console only uses heap data structures.  The actual details are stringified so that the resulting error is storable.